Mage.Sets/src/mage/cards/m/MangarasEquity.java

```diff
@@ -0,0 +1,161 @@
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.choices.ChoiceColor;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.targetpointer.FixedTarget;
+import mage.util.CardUtil;
+
+import java.util.UUID;
+
+/**
+ * @author Cguy7777
+ */
+public final class MangarasEquity extends CardImpl {
+
+    public MangarasEquity(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{1}{W}{W}");
+
+        // As Mangara's Equity enters the battlefield, choose black or red.
+        this.addAbility(new AsEntersBattlefieldAbility(new MangarasEquityColorChoiceEffect()));
+
+        // At the beginning of your upkeep, sacrifice Mangara's Equity unless you pay {1}{W}.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(
+                new SacrificeSourceUnlessPaysEffect(new ManaCostsImpl<>("{1}{W}")), TargetController.YOU, false));
+
+        // Whenever a creature of the chosen color deals damage to you or a white creature you control, Mangara's Equity deals that much damage to that creature.
+        this.addAbility(new MangarasEquityTriggeredAbility());
+    }
+
+    private MangarasEquity(final MangarasEquity card) {
+        super(card);
+    }
+
+    @Override
+    public MangarasEquity copy() {
+        return new MangarasEquity(this);
+    }
+}
+
+class MangarasEquityColorChoiceEffect extends OneShotEffect {
+
+    public MangarasEquityColorChoiceEffect() {
+        super(Outcome.Neutral);
+        staticText = "choose black or red";
+    }
+
+    private MangarasEquityColorChoiceEffect(final MangarasEquityColorChoiceEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Permanent permanent = game.getPermanentEntering(source.getSourceId());
+        if (permanent == null) {
+            return false;
+        }
+
+        ChoiceColor choice = new ChoiceColor(true, "Choose a color.", permanent);
+        choice.removeColorFromChoices("White");
+        choice.removeColorFromChoices("Blue");
+        choice.removeColorFromChoices("Green");
+
+        if (controller == null || !controller.choose(outcome, choice, game)) {
+            return false;
+        }
+
+        game.informPlayers(permanent.getLogName() + ": " + controller.getLogName() + " has chosen " + choice.getChoice());
+        game.getState().setValue(permanent.getId() + "_color", choice.getColor());
+        permanent.addInfo("chosen color", CardUtil.addToolTipMarkTags("Chosen color: " + choice.getChoice()), game);
+
+        return true;
+    }
+
+    @Override
+    public Effect copy() {
```

Comment:
covariant return: this should be `public MangarasEquityColorChoiceEffect copy()`